### PR TITLE
[FVM] Keeping event counter inside EventEmitter

### PR DESCRIPTION
**Pull request by Tonix517**
In PR #3268 the event counter was also moved into meter interface. It caused a problem in event index ([slack thread](https://dapperlabs.slack.com/archives/C01ANSMH5FF/p1665462723443849)) because meter itself has `enforceLimit` condition which event index should not be applied. Event index should increase unconditionally.
Fixed by keeping the event counter inside EventEmitter as usual.

Test
- [X] existing and patched UTs

### CODE REVIEW

**Review by pattyshack — APPROVED** [review]
since the event counter is not essential to meter limiting, should we just remove the eventCounter from meter?

**Comment by Tonix517** [comment]
addressed. good idea to keep our code lean.
> since the event counter is not essential to meter limiting, should we just remove the eventCounter from meter?

**Review by pattyshack — COMMENTED** [review]
lgtm

File: fvm/fvm_fuzz_test.go [hunk 1]
```diff
@@ -198,6 +198,12 @@ var fuzzTransactionTypes = []transactionType{
 			fees, deducted := getDeductedFees(t, tctx, results)
 			require.True(t, deducted, "Fees should be deducted.")
 			require.GreaterOrEqual(t, fees.ToGoValue().(uint64), fuzzTestsInclusionFees)
+			// event indices have to be an increasing uint sequence (0, 1, 2 ...)
+			expectedEventIndex := uint32(0)
+			for _, event := range results.tx.Events {
+				require.Equal(t, expectedEventIndex, event.EventIndex)
+				expectedEventIndex++
+			}
```
Comment (janezpodhostnik) [inline fvm/fvm_fuzz_test.go:206]: can you package this into a function and add it to all the test cases. (You can do that in a separate PR so this can be merged faster)

**Review by janezpodhostnik — APPROVED** [review]
Can you (in a separate PR or in this one) also add two more tests:
One testing that not only the fee deduction events, but also events emitted during transaction invocation are numbered correctly, and one testing that service events also respect the increasing index.

**Comment by Tonix517** [comment]
bors merge

**Comment by Tonix517** [comment]
bors retry
